index.js

```diff
@@ -40,3 +47,86 @@ server.get("/api/posts/:id", (req, res) => {
         .json({ error: "The post information could not be retrieved." });
     });
 });
+
+server.post("/api/posts", (req, res) => {
+  const postInfo = req.body;
+  if (!postInfo.title || !postInfo.contents) {
+    res.status(400).json({
+      errorMessage: "Please provide title and contents for the post."
+    });
+  } else {
+    db.insert(postInfo)
+      .then(postID =>
+        db
+          .findById(postID.id)
+          .then(post => res.status(201).json(post))
+          .catch(err =>
+            res.status(500).json({
+              error: "There was an error while saving the post to the database"
+            })
+          )
+      )
+      .catch(err =>
+        res.status(500).json({
+          error: "insert failed"
+        })
+      );
+  }
+});
+server.delete("/api/posts/:id", (req, res) => {
+  const { id } = req.params;
+  db.findById(id)
+    .then(post => {
+      if (post.length) {
+        db.remove(id)
+          .then(
+            res
+              .status(200)
+              .json({ message: `The post with the id ${id} has been deleted` })
+          )
+          .catch(err =>
+            res.status(500).json({ error: "The post could not be removed" })
+          );
+      } else {
+        res
+          .status(404)
+          .json({ message: "The post with the specified ID does not exist." });
+      }
+    })
+    .catch(err => res.status(500).json({ error: "findById failed" }));
+});
+server.put("/api/posts/:id", (req, res) => {
+  const { id } = req.params;
+  const { title, contents } = req.body;
+  db.findById(id)
+    .then(post => {
+      if (!post.length) {
+        res
+          .status(404)
+          .json({ message: "The post with the specified ID does not exist." });
+      } else if (title === undefined || contents === undefined) {
+        res.status(400).json({
+          errorMessage: "Please provide title and contents for the post."
+        });
+      } else {
+        db.update(id, { title, contents })
+          .then(count => {
+            db.findById(id)
```

Comment:
Update already looks the ID so I feel like this `findById` call is a little redundant.